### PR TITLE
feat: add otel collector and otlp support

### PR DESCRIPTION
**Pull request by beeme1mr**
## This PR

- adds updated OpenTelemetry hooks
- add the OpenTelemetry collector to Docker Compose
- add Prometheus

Signed-off-by: Michael Beemer <beeme1mr@users.noreply.github.com>

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -75,6 +75,31 @@ services:
...
+    command: [ "--config=/etc/otel-collector-config.yaml" ]
+    volumes:
+      - ./config/otel/collector.yaml:/etc/otel-collector-config.yaml
+    expose:
+      - "1888"   # pprof extension
+      - "8888"   # Prometheus metrics exposed by the collector
+      - "8889"   # Prometheus exporter metrics
+      - "13133" # health_check extension
+      - "4317"   # OTLP gRPC receiver
+      - "55679" # zpages extension
```
Comment (Kavindu-Dodan) [inline docker-compose.yaml:90]: minor - can be omitted

File: docker-compose.yaml [hunk 2]
```diff
@@ -75,6 +75,31 @@ services:
...
+    container_name: prometheus
+    image: prom/prometheus:latest
+    restart: always
+    volumes:
+      - ./config/prometheus/prometheus.yaml:/etc/prometheus/prometheus.yml
+    ports:
+      - "9090:9090"
+
   flagd:
     image: ghcr.io/open-feature/flagd:v0.5.4
```
Comment (Kavindu-Dodan) [inline docker-compose.yaml:104]: can bump to v0.6.0 which contain renamed metrics [1]

[1] - https://github.com/open-feature/flagd/releases/tag/flagd%2Fv0.6.0

File: scripts/tracing.js [hunk 3]
```diff
@@ -1,41 +1,42 @@
 'use strict';
 
-const { propagation } = require('@opentelemetry/api');
+const { NodeSDK } = require('@opentelemetry/sdk-node');
```
Comment (toddbaert) [inline scripts/tracing.js:3]: I'm a bit surprised this isn't just the API. I thought that consumers only needed to deal with that.

**Review by toddbaert — APPROVED** [review]

**Comment by toddbaert** [comment]
This is awesome, thanks.

File: scripts/tracing.js [hunk 4]
```diff
@@ -1,41 +1,42 @@
...
-
 const serviceName = process.env['OTEL_SERVICE_NAME'] || 'fib3r';
+const collectorUrl = process.env['OTEL_COLLECTOR_URL'] || 'http://otel-collector:4317';
 
-const provider = new NodeTracerProvider({
+const sdk = new NodeSDK({
   resource: new Resource({
     [SemanticResourceAttributes.SERVICE_NAME]: serviceName,
   }),
+  instrumentations: [getNodeAutoInstrumentations()],
```
Comment (Kavindu-Dodan) [inline scripts/tracing.js:26]: Related - https://github.com/open-feature/flagd/issues/754

Unfortunately this is the reason for the error logs

File: scripts/tracing.js (hunk 3 above)
Comment (beeme1mr) [reply scripts/tracing.js:3]: This is part of the instrumentation and exporter configuration.

**Pull request merged**